codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/BuilderGenerator.kt

```diff
@@ -217,17 +223,24 @@ class BuilderGenerator(
         docs("To override the contents of this collection use [`${member.setterName()}`](Self::${member.setterName()}).")
         rust("///")
         documentShape(member, model, autoSuppressMissingDocs = false)
+        val k = when (coreType.key) {
+            is RustType.String,
+            is RustType.Box -> "impl Into<${coreType.key.render(true)}>"
+            else -> "${coreType.key.render(true)}"
+        }
+        val v = when (coreType.member) {
+            is RustType.String,
+            is RustType.Box -> "impl Into<${coreType.member.render(true)}>"
+            else -> "${coreType.member.render(true)}"
+        }
+
         rustBlock(
-            "pub fn $memberName(mut self, k: impl Into<${coreType.key.render(true)}>, v: impl Into<${
-            coreType.member.render(
-                true
-            )
-            }>) -> Self"
+            "pub fn $memberName(mut self, k: $k, v: $v) -> Self"
         ) {
             rust(
                 """
                 let mut hash_map = self.$memberName.unwrap_or_default();
-                hash_map.insert(k.into(), v.into());
```

Comment:
probably need to conditionally generate the `.into()` invocations here

Reply:
yeah, based on this issue and the thing above I'll make a util that returns a tuple of the type and the value ie `["impl Into<${coreType.member.render(true)}>", "v.into()"]`